src/vpn-worker.ts

```diff
@@ -105,63 +105,62 @@ const worker = async (instanceId: number, serviceId: number) => {
 	});
 
 	const drainConnections = async () => {
-		const clientCount = Object.keys(clientCache).length;
+		let clientCount = Object.keys(clientCache).length;
 		logger.info(`${clientCount} clients connected`);
 
 		if (clientCount > 0) {
-			logger.debug(JSON.stringify(clientCache));
 			const delayMs = DEFAULT_SIGTERM_TIMEOUT / clientCount;
 			logger.info(
 				`disconnecting ${clientCount} clients, spaced by ${delayMs}ms`,
 			);
 
 			let timeToKill = DEFAULT_SIGTERM_TIMEOUT;
 
-			for (let clientId = 0; clientId < clientCount; clientId++) {
+			for (const cn of Object.values(clientCache)) {
 				try {
-					logger.debug(JSON.stringify(clientCache[clientId]));
-					const cn = clientCache[clientId].uuid;
-					logger.info(`disconnecting ${cn}`);
-					// update device state
-					clients.setConnected(cn, serviceId, false, logger);
-					// disconnect client from VPN
-					await vpn.killClient(cn);
+					try {
+						// disconnect client from VPN
+						try {
+							logger.info(`disconnecting ${cn}`);
+							await vpn.killClient(cn);
+						} catch (err) {
+							logger.warning(
+								`${err} while killing ${cn} on worker ${serviceId}`,
+							);
+						}
+
+						// update device state
+						clients.setConnected(cn, serviceId, false, logger);
+					} catch {
+						// https://stackoverflow.com/questions/3633362/is-there-any-way-to-use-a-numeric-type-as-an-object-key/3633390#3633390
+						// https://bobbyhadz.com/blog/typescript-element-implicitly-any-type-not-number
+						// https://www.typescriptlang.org/tsconfig#suppressImplicitAnyIndexErrors
```

Comment:
These seem to be related to the previous issue around using the object key from `Object.entries` so I'm assuming these comments are no longer necessary? (and probably the try/catch is also no longer necessary?)